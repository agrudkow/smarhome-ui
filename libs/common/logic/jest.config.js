module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
