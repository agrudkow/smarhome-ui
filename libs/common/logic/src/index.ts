export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
