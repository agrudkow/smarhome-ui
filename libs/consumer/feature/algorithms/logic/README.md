# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
