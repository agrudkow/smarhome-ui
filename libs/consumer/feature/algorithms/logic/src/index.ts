export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
