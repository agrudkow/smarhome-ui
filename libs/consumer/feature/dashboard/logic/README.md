# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
