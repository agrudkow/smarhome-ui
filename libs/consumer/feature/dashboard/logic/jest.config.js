module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
