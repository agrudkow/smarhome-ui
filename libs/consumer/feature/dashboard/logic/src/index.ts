export * from './lib/consumer-feature-dashboard-logic';
