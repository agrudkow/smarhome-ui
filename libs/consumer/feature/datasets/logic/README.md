# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
