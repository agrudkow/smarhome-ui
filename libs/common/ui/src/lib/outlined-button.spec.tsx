import React from 'react';
import { customRender } from '@smarthome/common/logic';
import OutlinedButton from './outlined-button';
import { fireEvent } from '@testing-library/react';

describe(' OutlinedButton', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <OutlinedButton onClick={jest.fn()} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <OutlinedButton onClick={jest.fn()} />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should call handler function on click', () => {
    const innerText = 'Button';
    const handleClick = jest.fn();
    const { getByText } = customRender(
      <OutlinedButton onClick={handleClick}>{innerText}</OutlinedButton>
    );
    fireEvent.click(getByText(innerText));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should be dissabled', () => {
    const innerText = 'Button';
    const { getByText } = customRender(
      <OutlinedButton onClick={jest.fn()} disabled>
        {innerText}
      </OutlinedButton>
    );
    expect(getByText(innerText).closest('button')).toBeDisabled();
  });

  it('should not be dissabled', () => {
    const innerText = 'Button';
    const { getByText } = customRender(
      <OutlinedButton onClick={jest.fn()}>{innerText}</OutlinedButton>
    );
    expect(getByText(innerText).closest('button')).not.toBeDisabled();
  });

  it('should set custom border color', () => {
    const borderColor = 'red';
    const { container } = customRender(
      <OutlinedButton onClick={jest.fn()} customBorderColor={borderColor} />
    );
    expect(container.firstChild).toHaveStyleRule('border-color', borderColor);
  });

  it('should set custom color', () => {
    const color = 'red';
    const { container } = customRender(
      <OutlinedButton onClick={jest.fn()} customColor={color} />
    );
    expect(container.firstChild).toHaveStyleRule('color', color);
  });
});
