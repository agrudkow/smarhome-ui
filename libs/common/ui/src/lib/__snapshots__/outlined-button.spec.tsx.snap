// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` OutlinedButton should match snapshot 1`] = `
.c0 {
  color: #555;
  border-color: #555;
  -webkit-transition: color 0.3s ease-in-out,border-color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out,border-color 0.3s ease-in-out;
  font-weight: 550;
  width: 100%;
}

.c0:hover {
  color: #1BBDCE;
  border-color: #1BBDCE;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined c0 MuiButton-outlinedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
